packages/shared/lib/core/wallet/actions/requestPersistedAsset.ts

```diff
@@ -1,12 +1,19 @@
+import { getErc20TokenMetadata } from '@core/layer-2'
+import { isValidEthereumAddress } from '@core/utils/crypto/utils'
 import { OFFICIAL_TOKEN_IDS } from '../constants'
 import { NotVerifiedStatus, VerifiedStatus } from '../enums'
 import { buildPersistedAssetFromIrc30Metadata } from '../helpers'
-import { IPersistedAsset } from '../interfaces'
+import { IErc20Metadata, IIrc30Metadata, IPersistedAsset } from '../interfaces'
 import { AssetVerification } from '../types'
-import { getIrc30MetadataFromFoundryOutput } from '../utils/getIrc30MetadataFromFoundryOutput'
+import { getIrc30MetadataFromFoundryOutput } from '../utils'
 
-export async function requestPersistedAsset(tokenId: string): Promise<IPersistedAsset> {
-    const tokenMetadata = await getIrc30MetadataFromFoundryOutput(tokenId)
+export async function requestPersistedAsset(tokenId: string, chainId?: number): Promise<IPersistedAsset | undefined> {
+    let tokenMetadata: IIrc30Metadata | IErc20Metadata | undefined
+    if (chainId && isValidEthereumAddress(tokenId)) {
```

Comment:
just to make sure: is this function really differentiating between erc20 ids and irc30 ids? both of them look like "0x...".
is the differentiating factor the checksum?

Reply:
an erc-20 token has as ID the smart contract address eg `0x639A647fbe20b6c8ac19E48E2de44ea792c62c5C`, whereas an irc20 token looks like `0x08d5e57a8b2adf771d475eb7c220efbb5ac4e70c40d33c691812c0e63feabe81a80100000000`, so the irc30 token id will fail the ethereum address check on both length and checksum